pupil_src/shared_modules/marker_tracker_3d/optimization/model_state.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import os
+
+import numpy as np
+
+import file_methods
+from marker_tracker_3d import utils
+
+logger = logging.getLogger(__name__)
+
+
+class ModelState:
+    def __init__(self, save_path):
+        self.save_path = save_path
+
+        self.current_frame_id = 0
+
+        self.frames_id = []
+        self.markers_id = []
+
+        self.all_novel_markers = []
+
+        self.camera_extrinsics_opt = {}  # {frame id: camera extrinsics(list)}
+        self.marker_extrinsics_opt = {}  # {marker id: marker extrinsics(list)}
+        self.marker_points_3d_opt = {}
+
+        self.model = None
+        self.load_marker_tracker_3d_model_from_file()
+
+    def load_marker_tracker_3d_model_from_file(self):
+        self.model = file_methods.Persistent_Dict(
```

Comment:
Is there a specific reason why you store `model` in the object? I guess you could just load the `Persistent_Dict`, store the values in `marker_extrinsics_opt` and forget about it? When you export it you have to fill the dict again anyway, so you can also create a new `Persistent_Dict`